backend/app/notify/email.go

```diff
@@ -0,0 +1,374 @@
+package notify
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net"
+	"net/smtp"
+	"sync"
+	"text/template"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+	"github.com/go-pkgz/repeater"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+)
+
+// EmailParams contain settings for email set up
+type EmailParams struct {
+	Host                 string        // SMTP host
+	Port                 int           // SMTP port
+	TLS                  bool          // TLS auth
+	From                 string        // From email field
+	Username             string        // user name
+	Password             string        // password
+	TimeOut              time.Duration // TLS connection timeout
+	MsgTemplate          string        // request message template
+	VerificationTemplate string        // verification message template
+	BufferSize           int           // email send buffer size
+	FlushDuration        time.Duration // maximum time after which email will me sent, 30s by default
+}
+
+// Email implements notify.Destination for email
+type Email struct {
+	EmailParams
+	smtpClient // initialized only on sending, closed afterwards
+
+	msgTmpl    *template.Template // parsed request message template
```

Comment:
`template.New` returns it and I thought it would be strange to convert it. I have no strong opinion here, please let me know if I need to unfold it from the pointer.

I can instead of storing it compile it every time during sending the message, but it would introduce unneeded computations and would delay the template compilation error, I would prefer to leave everything in the current state.

Reply:
> template.New returns it and I thought it would be strange to convert it.

makes sense